### PR TITLE
ciel: init at 3.1.3

### PR DESCRIPTION
**Pull request by YisuiDenghua**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ocfox — COMMENTED** [review]
I think cargo.lock should be added to the upstream repository instead of nixpkgs. Maybe you can communicate with the upstream.

**Comment by YisuiDenghua** [comment]
> I think cargo.lock should be added to the upstream repository instead of nixpkgs. Maybe you can communicate with the upstream.

Just removed Cargo.lock in this repo.

File: pkgs/tools/package-management/ciel/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
+  nativeBuildInputs = [ pkg-config zlib ];
+
+  doCheck = false;
+
+  buildInputs = [ systemd dbus openssl libssh2 libgit2 xz ];
+
+  cargoLock = {
+    lockFile = fetchurl {
```
Comment (NickCao) [inline pkgs/tools/package-management/ciel/default.nix:33]: This introduces IFD (which is not allowed in nixpkgs). Actually we ARE vendoring lock files into nixpkgs now, see: https://github.com/NixOS/nixpkgs/pull/217084

File: pkgs/tools/package-management/ciel/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  version = "3.1.3";
+
+  src = fetchFromGitHub {
+    owner = "AOSC-Dev";
+    repo = "ciel-rs";
+    rev = "02eac316765f27239b7b17176df812da702895cd";
+    hash = "sha256-WzrOCiIOdg3fBLKNjCPlr/XGrXC2424hO1nrwWXjx2A=";
+  };
+
+  nativeBuildInputs = [ pkg-config zlib ];
```
Comment (NickCao) [inline pkgs/tools/package-management/ciel/default.nix:26]: xz should be in buildInputs.

File: pkgs/tools/package-management/ciel/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    };
+  };
+
+  postInstall = ''
+    mv -v "$out/bin/ciel-rs" "$out/bin/ciel"
+  '';
+
+  meta = with lib; {
+    description = "A tool for controlling multi-layer file systems and containers.";
+    homepage = "https://www.rustup.rs/";
```
Comment (NickCao) [inline pkgs/tools/package-management/ciel/default.nix:48]: Copy paste error?

File: pkgs/tools/package-management/ciel/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    license = licenses.mit;
+    platforms = platforms.unix;
```
Comment (NickCao) [inline pkgs/tools/package-management/ciel/default.nix:50]: platforms.linux actually as systemd is not available on other platforms.

File: pkgs/tools/package-management/ciel/default.nix [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ciel";
+  version = "3.1.3";
+
+  src = fetchFromGitHub {
+    owner = "AOSC-Dev";
+    repo = "ciel-rs";
+    rev = "02eac316765f27239b7b17176df812da702895cd";
```
Comment (NickCao) [inline pkgs/tools/package-management/ciel/default.nix:22]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/tools/package-management/ciel/default.nix [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ciel";
+  version = "3.1.3";
+
+  src = fetchFromGitHub {
+    owner = "AOSC-Dev";
+    repo = "ciel-rs";
+    rev = "refs/tags/v{version}";
```
Comment (NickCao) [inline pkgs/tools/package-management/ciel/default.nix:23]: Typo here `${version}`

**Review by NickCao — COMMENTED** [review]
Overall LGTM. Please squash the commits.

File: pkgs/tools/package-management/ciel/default.nix [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  postPatch = ''
+    patchShebangs --build ./install-assets.sh
+  '';
+
+  postInstall = ''
+    mv -v "$out/bin/ciel-rs" "$out/bin/ciel"
+    env PREFIX="$out/" ./install-assets.sh
+    # FIXME: remove line below after https://github.com/AOSC-Dev/ciel-rs/pull/15 merged
+    mv -v $out/share/fish/completions $out/share/fish/vendor_completions.d
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/ciel/default.nix:48]: Please use installShellCompletion

**Comment by ocfox** [comment]
Please squash these commits as one.

**Comment by KiruyaMomochi** [comment]
Result of `nixpkgs-review pr 220006` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ciel</li>
  </ul>
</details>

**Comment by KiruyaMomochi** [comment]
Result of `nixpkgs-review pr 220006` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ciel</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]
checked build output.

**Pull request merged**